src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11AEADCipher.java

```diff
@@ -401,8 +401,27 @@ private void implInit(int opmode, Key key, byte[] iv, int tagLen,
     }
 
     private void cancelOperation() {
-        // cancel operation by finishing it; avoid killSession as some
-        // hardware vendors may require re-login
```

Comment:
The new `cancelOperation()` methods seems identical everywhere. Is it possible to consolidate it to a helper method like `trySessionCancel(token, session, flags)`? It can return true if canceled successfully, false if needs a fallback, and can still throw a `ProviderException`.

Reply:
I assume you mean the if-() block of code? I can move the code into a helper method inside the P11Util class.

Reply:
Yes, just keep duplicated lines as few as possible.